opslevel/terraform_type_conversions.go

```diff
@@ -17,6 +17,14 @@ func RequiredStringValue(value string) basetypes.StringValue {
 	return types.StringValue(unquote(value))
 }
 
+// Returns resourceValue wrapped in types.StringValue or types.StringNull if modelValue is null
+func StringValueFromResourceAndModelField(resourceValue string, modelValue basetypes.StringValue) basetypes.StringValue {
```

Comment:
should this be what `OptionalStringValue` does?

Reply:
The short answer is "it depends". I'm writing tests that will suss out all fields that have this issue. Then I'll apply this in a follow up PR